### PR TITLE
Datamodelling

### PR DESCRIPTION
**Pull request by mariebeersangvig**

### CODE REVIEW

File: pre-planning.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+----------------------------
+
+Identify the source systems that contains the necessary data to fill the
+master data requirements of the project.
+
+Identify their API, or other data protocol, and what connector can be
+used to communicate with them, including if the system has test and prod
+environments and their endpoint URIs.
+
+To be able to effectively impellent the project, a system owner and
```
Comment (grove) [inline pre-planning.rst:45]: Typo: impellent

File: archtecture-datamodelling.rst [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+This document introduces what we need to understand and consider before starting on a new project for a customer. So prior to start setting up infrastructure and doing work in Sesam we need to have a plan, or a data model showing how data flows from which systems, how it needs to be connected and transformed and which systems will receive data.
+
+Before we dive straight into data and dataflows we have to take a step back consider the following: there are a few risks that can affect the timeline of the pilot pr project phase, where both start date and end date can be delayed:
+
+• Clear scope of understanding of a pilot
+• Appointment of a Project Manager by a customer
+• Allocation of personnel by a customer
+• Scope and specifications of integrations 
+• Understanding of Sesam MasterData HUB
+
```
Comment (msauban) [inline archtecture-datamodelling.rst:21]: Should the document be more focused on the Sesam aspect of things rather than a general project setup? Those advices are very valid for any project, but seem to be outside the scope.

File: pre-planning.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+-  Product – Not screw, nut or bolt
+
+Connect data types to the globals
+---------------------------------
+
+Finally connect all identified data types to their respective globals to
+complete the data flow architecture.
+
+Booth business system data types and data platform data types may
```
Comment (grove) [inline pre-planning.rst:77]: Typo: Booth

**Comment by grove** [comment]
This PR does not build. The problem is that there is invalid syntax in the docs; incorrect links and headers primarily. The new docs also have to be added to the table of contents.

File: archtecture-datamodelling.rst [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+Considerations before setting up a new Sesam project
+----------------------------------------------------
+Testing
+=======
+
+The best practice for testing IT systems is involve testers as earliest as possible. This will ensure that test personnel will contribute to a specification phase and figure out how they should plan and test the requirements, both functional and non-functional. And also figuring what metrics they should use.
+This is regarded as the best practice by the most common used framework within Europe and US/Canada – ISTQB. And also aligned with a commonly used framework for specifying requirement used in Europe (required by EU entities) – IREB.
+By involving test at the earliest stage, one will ensure that all tests are planned for, and will be executed efficiently. And one could avoid the time-consuming exploratory test scenario.
```
Comment (msauban) [inline archtecture-datamodelling.rst:32]: similar to the previous comment, should we keep EU frameworks out?

File: archtecture-datamodelling.rst [hunk 5]
```diff
@@ -99,11 +143,46 @@ What you are trying to accomplish, is to have a set of global datasets that the
...
------------------------------------------------
+---------------------------------------------
 
 Azure
 =====
 
+Sesam with Azure components to form a data platform
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/azure_sesam.png
```
[the remark this answers is not in the document]
Comment (trondbbarstad) [reply archtecture-datamodelling.rst:154]: Yup, think so

File: archtecture-datamodelling.rst [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+Unit test
+^^^^^^^^^
+
+This is basically the developer´s node where he or she should test his newly developed feature or during development. This node is small, and one cannot expect to cover all possible test scenarios due to limitation on size of the node. This environment should contain modules and configurations, and data useful for development and personal unit test, and should not consist of other modules and configurations.
+
+System test
+^^^^^^^^^^^
+
+The first objective of a system test is not to see if a system has no errors, but to find errors. A good tester will cheer when a bug is discovered. As developers we should embrace this, it makes our software better. The second objective of a system test is to see if the system fulfills the stated requirements (the specified test cases and shots). If the test team participate in the specification phase, this will ensure that the requirements are testable and measurable. And the system test becomes efficiently for the functional test (just the test the requirements). 
+
```
Comment (msauban) [inline archtecture-datamodelling.rst:52]: While I agree it is important to explain terms that are going to be used in the document, it reads a little like a software engineering crash course. Should we go into details about what different types of tests are?

File: archtecture-datamodelling.rst [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+One of the purposes of a global dataset is to present a single authoritative truth about a concept or data. It is then logical to merge data from various different sources (or systems) in one global dataset if they define the same kind of object or type. For example, if some of the various sources contain person data, it would be logical to create a global dataset for person data and then merge each entity that refers to the same person. This is done so that when you ask for information about a specific entity, you also get information about that entity from the other systems. In terms of reusability this is a highly versatile way of getting all the data you need.
+
+However, merging data comes with a cost. In certain cases, changing the rules of how the data are merged requires the pipe to be reset and run again. For large datasets this might mean that it will take time before the downstream pipes will get updates.
+
+In some cases, merging the data isn’t logical. For instance, data like countries, counties, cities and streets might be put into a global location dataset, but it is not logical to merge these data.
```
Comment (msauban) [inline archtecture-datamodelling.rst:130]: Why is it not logical to merge location data? Maybe we could add what can make the example illogical.

File: archtecture-datamodelling.rst [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+To the point: Data Modelling
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let us begin by repeating an important truth about data modelling in Sesam: when modelling in Sesam do not create your data models by how they are related to each other by their properties (as in a standard relational database), but by what the data in the datasets are about.
+
+For example, let us say we have two datasets or tables, ‘employee and ‘child’. In a relational database, these two tables would normally be linked by a one-to-many relation, i.e. an employee can have many children. However, when we look at these two tables from a Sesam perspective, both are in general speaking of a ‘person’. Hence, we should put both the ‘employee’ and ‘child’ table in the same global containing data about a person, ‘global-person’.
+
+What you are trying to accomplish, is to have a set of global datasets that the preparation pipes can choose from, like food items grouped together in a supermarket to easily locate the food items you need. On the other hand, you don’t want the number of global datasets to grow out of hand, making it hard to find.
```
Comment (msauban) [inline archtecture-datamodelling.rst:141]: Food stalls is a good analogy, we can throw in the recipe to tailor the output to specific need.

**Review by msauban — COMMENTED** [review]
interesting read, sometimes maybe to generic, I can expand on that orally.
